server/book/orderpq.go

```diff
@@ -81,35 +82,22 @@ func (pq *OrderPQ) copy(newCap uint32) *OrderPQ {
 	return newPQ
 }
 
-// Orders copies all orders, sorted with the lessFn. To avoid modifying the
-// OrderPQ or any of the data fields, a deep copy of the OrderPQ is made and
-// ExtractBest is called until all entries are extracted.
+// Orders copies all orders, sorted with the lessFn. The OrderPQ is unmodified.
 func (pq *OrderPQ) Orders() []*order.LimitOrder {
+	// Deep copy the orders.
 	pq.mtx.RLock()
-
-	// To use the configured lessFn, make a temporary OrderPQ with just the
-	// lessFn and orderHeap set. Do not use the constructor since we do not care
-	// about the orders map or capacity.
-	pqTmp := OrderPQ{
-		lessFn: pq.lessFn,
-		oh:     make(orderHeap, len(pq.oh)),
+	orders := make([]*order.LimitOrder, len(pq.oh))
+	for i, oe := range pq.oh {
+		orders[i] = oe.order
 	}
-	copy(pqTmp.oh, pq.oh)
```

Comment:
This was bad because the `orderEntry`s' heapIdx was modified in the sort below.  It needed to be a deep copy.